packages/liveblocks-core/src/crdts/LiveObject.ts

```diff
@@ -544,6 +560,32 @@ export class LiveObject<O extends LsonObject> extends AbstractCrdt {
    */
   update(patch: Partial<O>): void {
     this._pool?.assertStorageIsWritable();
+
+    // If detectLargeObjects is enabled, perform a runtime size check now so we
+    // can immediately throw as soon as the max object size is exceeded.
+    if (LiveObject.detectLargeObjects) {
+      const data: Record<string, Json> = {};
+      for (const [key, value] of this.#map) {
+        if (!isLiveNode(value)) {
+          data[key] = value;
+        }
+      }
+      for (const key of Object.keys(patch)) {
+        const value = patch[key];
+        if (value === undefined) continue;
+        if (!isLiveNode(value)) {
+          data[key] = value;
+        }
+      }
+
+      const newSize = new TextEncoder().encode(JSON.stringify(data)).length;
```

Comment:
We're reading the number of bytes here but I guess `.length` returns the same value as `.byteLength` on a `Uint8Array`, so this is okay!